kratos/python/add_matrix_to_python.cpp

```diff
@@ -89,6 +89,25 @@ namespace Python
 
           return matrix;
         }));
+
+        matrix_binder.def(py::init([](const py::list& input){
+          std::size_t num_rows = input.size();
+          if( num_rows == 0 || ( (num_rows == 1) && ((py::cast<py::list>(input[0])).size() == 0) ) )
```

Comment:
can you please try without the cast?

if it doesn't work please post the issue

Reply:
Yes, It can be done with `pybind11::len()` function